controller/metrics/rollouts.go

```diff
@@ -136,6 +143,7 @@ func collectRollouts(ch chan<- prometheus.Metric, rollout *v1alpha1.Rollout) {
 
 	addGauge(descRolloutInfoReplicasAvailable, float64(rollout.Status.AvailableReplicas))
 	addGauge(descRolloutInfoReplicasUnavailable, float64(rollout.Status.Replicas-rollout.Status.AvailableReplicas))
+	addGauge(descRolloutInfoReplicasDesired, float64(*rollout.Spec.Replicas))
```

Comment:
There is a chance, albeit small, for this to have a nil pointer dereference if the the endpoint is scraped from the time a rollout is created, to the time the controller was able to default the replicas to 1.

Could you replace `*rollout.Spec.Replicas` with:

```
defaults.GetReplicasOrDefault(rollout.Spec.Replicas)
```

in "github.com/argoproj/argo-rollouts/utils/defaults" package?